chips/sam4l/src/usbc/common_register.rs

```diff
@@ -0,0 +1,244 @@
+#![allow(dead_code)]
+
+use core::marker::PhantomData;
+
+/// A memory-mapped register
+#[derive(Copy, Clone)]
+pub struct Reg<T> {
+    addr: *mut u32,
```

Comment:
Why not just `*mut T`. Then you don't need the `PhantomData` field or any of the `ToWord/FromWord` stuff